openmc/model/funcs.py

```diff
@@ -111,6 +112,151 @@ def borated_water(boron_ppm, temperature=293., pressure=0.1013, temp_unit='K',
+        'upper_fillet_radius',
+        upper_fillet_radius,
+        r,
+        equality=True)
+    check_type('lower_fillet_radius', lower_fillet_radius, Real)
+    check_less_than(
+        'lower_fillet_radius',
+        lower_fillet_radius,
+        r,
+        equality=True)
+    check_value('axis', axis, ['x', 'y', 'z'])
+    check_type('origin', origin, Iterable, Real)
+
+    # Define function to create a plane on given axis
+    def plane(axis, name, value):
+        cls = getattr(openmc, f'{axis.upper()}Plane')
+        return cls(value, name=f'{name} {axis}',
+                   boundary_type=boundary_type)
+
+    def cylinder(axis, name, ax1, val1, ax2, val2, r):
+        cls = getattr(openmc, f'{axis.upper()}Cylinder')
+        return cls(name=f'{name} {axis}',
+                   boundary_type=boundary_type,
+                   **{ax1 + '0': val1,
+                      ax2 + '0': val2,
+                      'r': r})
+
+    if axis == 'x':
+        x1, x2 = 'y', 'z'
+        axcoord, axcoord1, axcoord2 = 0, 1, 2
+    elif axis == 'y':
+        x1, x2 = 'x', 'z'
+        axcoord, axcoord1, axcoord2 = 1, 0, 2
+    else:
+        x1, x2 = 'x', 'y'
+        axcoord, axcoord1, axcoord2 = 2, 0, 1
+
+    # Create cylindrical region
+    min_h = plane(axis, 'minimum', -height / 2 + origin[axcoord])
+    max_h = plane(axis, 'maximum', height / 2 + origin[axcoord])
+    radial = cylinder(axis, 'outer',
+                      x1, origin[axcoord1],
+                      x2, origin[axcoord2],
+                      r)
+
+    if boundary_type == 'periodic':
+        min_h.periodic_surface = max_h
+        max_h.periodic_surface = min_h
+    prism = +min_h & -max_h & -radial
+
+    # Handle rounded corners if given
+    if upper_fillet_radius > 0. or lower_fillet_radius > 0.:
+        # Get torus class corresponding to given axis
+        tor = globals()['{}Torus'.format(axis.upper())]
+
+        if boundary_type == 'periodic':
+            raise ValueError('Periodic boundary conditions not permitted when '
+                             'rounded corners are used.')
+
+        corners_upper = None
+        corners_lower = None
+        args = {
+            'boundary_type': boundary_type,
+            x1 + '0': origin[axcoord1],
+            x2 + '0': origin[axcoord2]
+        }
+
+        def create_fillet(args, r, fillet_radius, pos):
+            args['a'] = r - fillet_radius
+            args['b'] = fillet_radius
+            args['c'] = fillet_radius
+            if pos:
+                coord = origin[axcoord] + height / 2 - fillet_radius
+                tor_name = f'{axis} max'
+                cyl_name = 'upper_min'
+                axial_bound = +plane(axis, 'upper min', coord)
+            else:
+                coord = origin[axcoord] - height / 2 + lower_fillet_radius
+                tor_name = f'{axis} min'
+                cyl_name = 'lower_min'
+                axial_bound = -plane(axis, 'lower max', coord)
```

Comment:
I was thinking something like this, but not sure if it's
1. 100% valid (can surface sense be changes as a multiply operation?)
2. too clever? Is it harder to understand for future developers?
3. messing up surface naming conventions?


```suggestion
        def create_fillet(args, r, fillet_radius, pos='upper'):
            args['a'] = r - fillet_radius
            args['b'] = fillet_radius
            args['c'] = fillet_radius
            fillet_ext = height / 2 - fillet_radius
            tor_name = f'{axis} {pos}'
            cyl_name = f'{pos}_min'
            dir = 1
            if pos == 'lower':
                dir = -1
            coord = origin[axcoord] + dir * fillet_ext
            axial_bound = dir * plane(axis, f'{pos} ext', coord)
```

Reply:
I updated your suggestion 

> 1. 100% valid (can surface sense be changes as a multiply operation?)
> 
> 2. too clever? Is it harder to understand for future developers?
> 
> 3. messing up surface naming conventions?

1. It's wonderful, but we can't multiply `int` by a `Surface` to get what we want.
2. I don't think so.
3. Nope.

I modified your original suggestion a bit so it works.